panic in the TestStateSyncerSimple

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
 === RUN   TestSyncerSimple
[2023/05/18 02:53:25.758 +08:00] [INFO] [manager.go:178] ["start campaign owner"] [ownerInfo="[ddl] /tidb/ddl/fg/owner"]
panic: Fail in goroutine after TestStateSyncerSimple has completed
goroutine 292 [running]:
testing.(*common).Fail(0xc00066eea0)
	GOROOT/src/testing/testing.go:933 +0x1a8
testing.(*common).Errorf(0xc00066eea0, {0x4021df6, 0x3}, {0xc000a13500, 0x1, 0x1})
	GOROOT/src/testing/testing.go:1050 +0xaf
github.com/stretchr/testify/assert.Fail({0x7f8c2edc1bd8, 0xc00066eea0}, {0xc000dc6240, 0x225}, {0x0, 0x0, 0x0})
	external/com_github_stretchr_testify/assert/assertions.go:262 +0x412
github.com/stretchr/testify/assert.NoError({0x7f8c2edc1bd8, 0xc00066eea0}, {0x588d3e0, 0xc000139578}, {0x0, 0x0, 0x0})
	external/com_github_stretchr_testify/assert/assertions.go:1380 +0x127
github.com/stretchr/testify/require.NoError({0x589ad80, 0xc00066eea0}, {0x588d3e0, 0xc000139578}, {0x0, 0x0, 0x0})
	external/com_github_stretchr_testify/require/require.go:1261 +0xa5
ddl/syncer/syncer_test_test.TestStateSyncerSimple.func3()
	ddl/syncer/state_syncer_test.go:66 +0x74
created by ddl/syncer/syncer_test_test.TestStateSyncerSimple
	ddl/syncer/state_syncer_test.go:65 +0x6aa 
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/ddl/syncer/state_syncer_test.go
+++ b/ddl/syncer/state_syncer_test.go
@@ -62,9 +62,10 @@ func TestStateSyncerSimple(t *testing.T) {
 		WithLease(testLease),
 		WithInfoCache(ic),
 	)
-	go func() {
+	var wg util.WaitGroupWrapper
+	wg.Run(func() {
 		require.NoError(t, d.OwnerManager().CampaignOwner())
-	}()
+	})
 	defer d.OwnerManager().Cancel()
 	// TODO: We can remove it when we call it in newDDL.
 	require.NoError(t, d.StateSyncer().Init(ctx))
@@ -77,7 +78,6 @@ func TestStateSyncerSimple(t *testing.T) {
 	require.Equal(t, stateInfo, respState)
 	require.False(t, d.StateSyncer().IsUpgradingState())
 	// for watchCh
-	var wg util.WaitGroupWrapper
 	var checkErr string
 	stateInfo.State = syncer.StateUpgrading
 	stateInfoByte, err := stateInfo.Marshal()

